package-res/resources/web/pentaho/type/instance.js

```diff
@@ -0,0 +1,274 @@
+       *
+       * @param {?boolean} [keyArgs.includeType=false] - Includes the inline type property, `_`, in the specification.
+       *
+       * @return {!any} A specification of this instance.
+       */
+      toSpec: function(keyArgs) {
+        return O.using(new SpecificationScope(), this.toSpecInContext.bind(this, keyArgs || {}));
+      },
+
+      /**
+       * Creates a specification that describes this instance.
+       *
+       * @param {Object} [keyArgs] - The keyword arguments object.
+       * Passed to every instance and type serialized within this scope.
+       *
+       * Please see the documentation of subclasses for information on additional, supported keyword arguments.
+       *
+       * @return {!any} A specification of this instance.
+       *
+       * @abstract
+       *
+       * @see pentaho.type.Instance#toSpec
+       */
+      toSpecInContext: function(keyArgs) {
+        /* istanbul ignore next : abstract method */
+        throw error.notImplemented();
+      },
+
+      /**
+       * Creates a top-level JSON specification that describes this instance.
+       *
+       * Attributes which don't have a JSON-compatible specification are omitted.
+       * Specifically, for inline types, attributes with a function value are not supported.
+       *
+       * This method simply calls {@link @see pentaho.type.Instance#toSpec} with argument `keyArgs.isJson` as `true`
+       * and exists for seamless integrations with JavaScript's
+       * [JSON.stringify](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)
+       * method.
+       *
+       * @see pentaho.type.Instance#toSpec
+       *
+       * @return {UJsonValue} A JSON-compatible specification.
+       */
+      toJSON: function() {
+        return this.toSpec({isJson: true});
+      }
+      //endregion
+    }, /** @lends pentaho.type.Instance */{
+
+      //region type property
+      /**
+       * Gets the type of this instance constructor.
+       *
+       * @type pentaho.type.Type
+       * @readonly
+       */
+      get type() {
+        return this.prototype.type;
+      },
+
+      // Supports Type class-level configuration only.
+      // Not documented on purpose.
+      // Allows writing;
+      //   Instance.implementStatic({type: .})
+      // to mean:
+      //   Type#implementStatic(.).instance.constructor
+      set type(config) {
+        if(config) this.type.constructor.implementStatic(config);
+      }, //endregion
+
+      // override the documentation to specialize the argument types.
+      /**
+       * Creates a subtype of this one.
+       *
+       * For more information on class extension, in general,
+       * see {@link pentaho.lang.Base.extend}.
+       *
+       * @name extend
+       * @memberOf pentaho.type.Instance
+       *
+       * @param {string} [name] The name of the created class. Used for debugging purposes.
```

Comment:
Suggest:

The name of the created class. The name of the created class is used for debugging purposes.